### PR TITLE
iproute2: update to 5.19 and shrink ip-tiny-size

### PR DESCRIPTION
**Pull request by PolynomialDivision**
The first two commits are fine to be merged. **However, in version 5.18.0 the ip-tiny size grows from 124k to 128k.**

### CODE REVIEW

**Comment by trippleflux** [comment]
Getting the following when builiding OpenWrt Snapshot x86_64 : 
```
make[3]: Entering directory '/home/username/works/openwrt/package/network/utils/iproute2'
mkdir -p /home/username/works/openwrt/bin/targets/x86/64-glibc/packages /home/username/works/openwrt/build_dir/target-x86_64_glibc_custom/linux-x86_64/iproute2-ipfull/iproute2-5.18.0/ipkg-x86_64/ss/CONTROL /home/username/works/openwrt/staging_dir/target-x86_64_glibc_custom/pkginfo
install -d -m0755 /home/username/works/openwrt/build_dir/target-x86_64_glibc_custom/linux-x86_64/iproute2-ipfull/iproute2-5.18.0/ipkg-x86_64/ss/usr/sbin
install -m0755 /home/username/works/openwrt/build_dir/target-x86_64_glibc_custom/linux-x86_64/iproute2-ipfull/iproute2-5.18.0/misc/ss /home/username/works/openwrt/build_dir/target-x86_64_glibc_custom/linux-x86_64/iproute2-ipfull/iproute2-5.18.0/ipkg-x86_64/ss/usr/sbin/
find /home/username/works/openwrt/build_dir/target-x86_64_glibc_custom/linux-x86_64/iproute2-ipfull/iproute2-5.18.0/ipkg-x86_64/ss -name 'CVS' -o -name '.svn' -o -name '.#*' -o -name '*~'| xargs -r rm -rf
Package ss is missing dependencies for the following libraries:
libtirpc.so.3
make[3]: *** [Makefile:287: /home/username/works/openwrt/bin/packages/x86_64/base/ss_5.18.0-2_x86_64.ipk] Error 1
make[3]: Leaving directory '/home/username/works/openwrt/package/network/utils/iproute2'
time: package/network/utils/iproute2/ipfull/compile#0.54#0.14#0.60
    ERROR: package/network/utils/iproute2 failed to build (build variant: ipfull).
make[2]: *** [package/Makefile:116: package/network/utils/iproute2/compile] Error 1
make[2]: Leaving directory '/home/username/works/openwrt'
make[1]: *** [package/Makefile:110: /home/username/works/openwrt/staging_dir/target-x86_64_glibc_custom/stamp/.package_compile] Error 2
make[1]: Leaving directory '/home/username/works/openwrt'
make: *** [/home/username/works/openwrt/include/toplevel.mk:230: world] Error 2
```

**Comment by PolynomialDivision** [comment]
@trippleflux Thanks for testing. Can you check again? I updated the PR and add a commit description why this happened.

**Comment by PolynomialDivision** [comment]
Oh wait I think that was a stupid solution. I also tried to override the HAVE_RPC variable, but it did not work. :/

File: package/network/utils/iproute2/Makefile [hunk 1]
```diff
@@ -100,7 +100,7 @@ endef
 define Package/ss
 $(call Package/iproute2/Default)
   TITLE:=Socket statistics utility
-  DEPENDS:=+libnl-tiny +(PACKAGE_devlink||PACKAGE_rdma):libmnl +kmod-netlink-diag
+  DEPENDS:=+libnl-tiny +PACKAGE_libtirpc:libtirpc +(PACKAGE_devlink||PACKAGE_rdma):libmnl +kmod-netlink-diag
```
Comment (hauke) [inline package/network/utils/iproute2/Makefile:103]: Why is `+PACKAGE_libtirpc:libtirpc` still needed when you set `HAVE_TIRPC=n`?

Comment (PolynomialDivision) [reply package/network/utils/iproute2/Makefile]: Did forget to delete it again. Thanks for spotting this.

**Comment by hauke** [comment]
I think a size increase of 4KB on ip-tiny should be ok. Maybe we can deactivate some other protocols in ip-tiny which are not commonly needed.

**Comment by PolynomialDivision** [comment]
> I think a size increase of 4KB on ip-tiny should be ok. Maybe we can deactivate some other protocols in ip-tiny which are not commonly needed.

I pushed today a update to 5.19 again increasing the size by around 4 KiB. : / I will look again what i can do to decrease size again.

**Comment by trippleflux** [comment]
@PolynomialDivision 
Thank you for the proper fixes, 5.19 deployed in x86_64, so far no problem.

**Comment by PolynomialDivision** [comment]
@hauke I removed now some features bringing it back to 124k on mips.

**Comment by Neustradamus** [comment]
@PolynomialDivision: I think that you must edited the main description because you have reduced the code...

**Comment by hauke** [comment]
@PolynomialDivision  thanks for removing some modules. Did you check if they are used anywhere?

**Comment by PolynomialDivision** [comment]
@hauke Correct me if I'm wrong, but I strongly believe that the busybox ip is normally shipped in default openwrt images. And I did not find any package that depends on `ip-tiny`, neither in packages, main or routing.

**Comment by hauke** [comment]
Thank you for the patches, I applied them to master in 991351431998698aa3d43c9578b8a5184794d1da, e65337ce65eb5487f10895b75e93002983aa6194, d3b4422f6206968d17bbf42cfb911c8ff699b160, e87114401357ccd34893232b35a757d4c33ea710 and b3a0c14824be04b40e5098f3dc5073bb4b250edf.

**Pull request closed without merging**